src/hotspot/cpu/riscv/macroAssembler_riscv.cpp

```diff
@@ -1167,6 +1167,19 @@ void MacroAssembler::fsflagsi(Register Rd, unsigned imm) {
 
 #undef INSN
 
+void MacroAssembler::restore_cpu_control_state_after_jni(Register tmp1, Register tmp2) {
+  if (RestoreMXCSROnJNICalls) {
+    Label skip_fsrmi;
+    frrm(tmp1);
+    mv(tmp2, RoundingMode::rne);
+    // Set FRM to the state we need. We do want Round to Nearest. We
+    // don't want non-IEEE rounding modes.
+    beq(tmp1, tmp2, skip_fsrmi);        // Only reset FRM if it's wrong
```

Comment:
Same as https://github.com/openjdk/jdk/pull/18758#discussion_r1566037120, you can take advantage of `RoundingMode::rne == 0` by doing `beq(tmp1, zr, skip_fsrmi);` and adding a `guarantee(RoundingMode::rne == 0);` right above for self-documentation.